src/prpy/util.py

```diff
@@ -1326,7 +1326,7 @@ def ComputeGeodesicUnitTiming(traj, env=None, alpha=1.0):
     return new_traj
 
 
-def CheckJointLimits(robot, q, deterministic=None):
+def CheckJointLimits(robot, q, hard_limits, deterministic=None):
```

Comment:
This is a little weird because the only reason we store the soft limits is because we are using OpenRAVE's `GetActiveDOFLimits()` function, which isn't integral to `prpy`.  So adding the hard limits is not quite analogous, even if we put it in the `prpy.base.Robot`.

Can someone bring me up to speed on what the usage pattern for soft vs. hard limits is intended to be?